membrane_curvature/base.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+from membrane_curvature.lib.mods import derive_surface, mean_curvature, gaussian_curvature
+import numpy as np
+
+
+def __init__(self, universe, select,
+             x_bins=100, y_bins=100,
+             x_range=None,
+             y_range=None,
+             pbc=True):
+    self.universe = universe
+    self.selection = select
+    self.pbc = pbc
+    self.y_bins = y_bins
+    self.x_bins = x_bins
+    self.x_range = x_range if x_range else (0, universe.dimensions[0])
+    self.y_range = y_range if y_range else (0, universe.dimensions[1])
+
+    # Warns if range doesn't cover entire dimensions of simulation box
+    if (x_range < universe.dimensions[0]) or (y_range < universe.dimensions[1]):
+        raise ValueError("Minimum range must be ({}, {})").format(self.universe.dimensions[0],
+                                                                  self.universe.dimenions[1])
+
+
+def _prepare(self):
+
+    # Initialize empty np.array(cumulative, count) of results
+    cumulative = np.zeros((self.x_bins, self.y_bins))
+    count = np.zeros((self.x_bins, self.y_bins))
+
+    self.results.surface = (cumulative, count)
+    self.results.mean = (cumulative, count)
+    self.results.gaussian = (cumulative, count)
+
+
+def _output(self, value_per_frame, cumulative_output, counter):
```

Comment:
I wanted to ask you something about this approach, @lilyminium . To be honest, I didn't see any other sensible option to accumulate the values per frame and count. Would you please share with me some alternatives? What do you think would be a good idea? 
(Not that I am looking for a change ☺️ , more to consider alternatives I can't see by myself)

Reply:
Do you need to accumulate the count per frame as an end goal? My first thought would be to collect all the surfaces in a 3D array: (n_frames, n_x_cells, n_y_cells) and then average over them with `np.nanmean` for the average surface.